homework4/4.2/4.2/List.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Task4_2.Exception;
+
+namespace Task4_2;
+
+/// <summary>
+/// Список элементов
+/// </summary>
+public class List
+{
+    /// <summary>
+    /// Элемент списка, содержит в себе значение и ссылку на следующий элемент
+    /// </summary>
+    private class ListElement
+    {
+        public ListElement(int value, ListElement next)
+        {
+            Value = value;
+            Next = next;
+        }
+
+        public ListElement Next { get; set; }
+        public int Value { get; set; }
+    }
+    /// <summary>
+    /// Первый элемент списка
+    /// </summary>
+    private ListElement head;
```

Comment:
```suggestion
    private ListElement? head;
```
Список может быть пустым, следовательно голове разрешено быть null